File: elearning-service/src/main/java/com/app/elearningservice/dto/LessonDTO.java
package com.app.elearningservice.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class LessonDTO {
    private Long lessonId;
    private String title;
    private String description;
    private Integer sequence;

    private Long mediaId;
    private String mediaTitle;
    private String url;
    private String type;
}

File: elearning-ui/src/app/shared/model/LessonDTO.ts
export class LessonDTO {
  constructor(
    public lessonId: number = -1,
    public courseId: number = -1,
    public title: string = '',
    public description: string = '',
    public sequence: number = 0,
    public expanded: boolean = false,
    public media: MediaLesson[] = []
  ) {
  }
}

export class MediaLesson {
  constructor(
    public mediaId: number = 0,
    public lessonId: number = 0,
    public title: string = '',
    public url: string = '',
    public urlBlob: any = null,
    public type: string = '',
  ) {
  }
}
